client.go

```diff
@@ -858,3 +888,77 @@ func (c *Client) SetMaxBytesToParse(max int) *Client {
 	c.maxBytesToParse = max
 	return c
 }
+
+// getIpFromInterface return a IPv4 and IPv6 address from the interface.
+// if there is no address than that IPfamily is nil.
+func getIpFromInterface(theInterface string) (theIPs localTCPIP) {
+	if theInterface == "" {
+		return
+	}
+	ifaces, err := netInterfaces()
+	if err != nil {
+		log.Errorf("getIpFromInterface: %+v\n", err.Error())
+		return
+	}
+	log.Debugf("netInterfaces: %+v\n", ifaces)
+	for _, i := range ifaces {
+		if i.Name != theInterface {
+			continue
+		}
+		addrs, err := netInterfaceAddrs(&i)
+		if err != nil {
+			log.Errorf("getIpFromInterface: %+v\n", err.Error())
+			continue
+		}
+		for _, a := range addrs {
+			log.Debugf("Addr : %+v\n", a)
+			switch v := a.(type) {
+			case *net.IPAddr:
```

Comment:
Besides the debug log the 2 branches are the same.

Reply:
If both net.IPAddr and net.IPNet is in the same case, then v := a.(type) is not working what cause that v doesn't have IP field.